test/y2storage/initial_guided_proposal_test.rb

```diff
@@ -162,22 +164,51 @@
       end
 
       context "and a proposal is not possible with any individual device" do
+        let(:swap_optional) { false }
+
         before do
           sda.size = 12.GiB
           sdb.size = 15.GiB
           sdc.size = 3.GiB
         end
 
-        it "uses all the devices to make the proposal" do
+        it "allocates the root device in the biggest device" do
           proposal.propose
 
-          expect(used_devices).to contain_exactly("/dev/sda", "/dev/sdb", "/dev/sdc")
+          expect(disk_for("/").name).to eq "/dev/sdb"
         end
 
-        it "allocates the root device in the biggest device" do
-          proposal.propose
+        # FIXME: this is currently not working, see bsc#1130392
```

Comment:
Ups, this is a quite important bug. After disabling a volume feature (snapshots, adjust_by_ram, or the volume itself) the disabled feature is never restored. Good catch!

Definitely, [`Yast.deep_copy`](https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/initial_guided_proposal.rb#L41) does not work as I expected.

Reply:
@joseivanlopez can you get more details how and why deep code does not work as expected?

Reply:
Well, somehow I supposed a different behavior (I don't know why because Yast.deep_copy is well documented). What we need here it is something like the following (a recursive version):

```
class Foo 
    attr_accessor :bar

    def initialize
        @foobar = [2, 2]
    end 

    def deep_copy
        f = dup 

        instance_variables.each do |v| 
            value = instance_variable_get(v)
            f.instance_variable_set(v, value.dup)
        end 

        f   
    end 

private

    attr_accessor :foobar
end

f = Foo.new
f.bar = [1, 1]

f2 = f.deep_copy
f2.bar << 1
f2.instance_variable_get(:@foobar) << 2

f.send(:bar) #=> [1, 1]
f2.send(:bar) #=> [1, 1, 1]

f.send(:foobar) #=> [2, 2]
f2.send(:foobar) #=> [2, 2, 2]
```

Do you think something like this Foo#deep_copy is a good idea?

Reply:
Well, problem is that deep_copy is intended only for old component system. If you have own class I suggest to use common ruby approach ( whatever it is ). Maybe just overwrite dup?

Reply:
Yes, I guess we can add `ProposalSettings#dup` with a logic similar as my example.